### PR TITLE
Add make sprint-data-import and issue-data-import to import github sprint and issue data to database

### LINKED ISSUE #46: [Task]:  Add `make sprint-data-import` and `issue-data-import` to import github sprint and issue data to database
### Summary

From: https://github.com/HHS/simpler-grants-gov/issues/1600

One strategy would be to keep the  `make sprint-data-export` and `issue-data-export` and create  `make sprint-db-data-import` and `issue-data-db-import` so that the data is exported to JSON and then imported into the database.

A single make command could then be created to run the the export and then import files.

Bonus points if there is a way to pipe the Github data directly to the database.

### Acceptance criteria

_No response_

### PR DESCRIPTION
**Pull request by aplybeah**
## Summary
Fixes #46 

## Changes proposed
* added `sprint-db-data-import` to Makefile
* added `export_json_to_database`
## Context for reviewers

> One strategy would be to keep the make sprint-data-export and issue-data-export and create make sprint-db-data-import and issue-data-db-import so that the data is exported to JSON and then imported into the database.
> 
> A single make command could then be created to run the the export and then import files.

## Additional information
Sample data in database

Screen Shot 2024-06-26 at 3 38 47 PM

### CODE REVIEW

File: analytics/config.py [hunk 1]
```diff
@@ -21,8 +21,8 @@
     merge_enabled= True,
     # add validators for our required config vars
     validators=[
-        Validator("SLACK_BOT_TOKEN", must_exist=True),
-        Validator("REPORTING_CHANNEL_ID", must_exist=True),
+        Validator("SLACK_BOT_TOKEN", must_exist=False), #disabled for testing
+        Validator("REPORTING_CHANNEL_ID", must_exist=False), #disabled for testing
```
Comment (coilysiren) [inline analytics/config.py]: Billy wrote some end to end tests that require these tokens. So these need to stay enabled if we want to run the full end to end test suite, unfortunately.

That said, I don't think we should have ever written end to end tests that require these tokens to be set. So I'm willing to accept this change.

File: analytics/Makefile [hunk 2]
```diff
@@ -146,6 +146,17 @@ sprint-data-export:
 	--project $(SPRINT_PROJECT) \
 	--output-file $(SPRINT_FILE)
 
+# Import file data to database
+# Takes sprint and issue and puts it in database
+sprint-db-data-import:
+	@echo "=> Importing project data to the database"
+	@echo "====================================================="
+	$(POETRY) analytics export db_export \
```
Comment (coilysiren) [inline analytics/Makefile]: ⭐⭐⭐  &nbsp; _(must change before approval)_ The makefile command says db-**import** whereas the python command is db-**export**. You'll want to pick one or the other. I'm thinking import?

**Comment by acouch** [comment]
Looks good, can you fix the lint test?

**Comment by acouch** [comment]
When I run this locally I get:

```bash

(analytics)(import-sprint-data)$make sprint-db-data-import
=> Importing project data to the database
=====================================================
docker-compose run -e GH_TOKEN --rm grants-analytics poetry run analytics export db_export \
	--owner HHS \
	--project 13 \
	--sprint-file data/sprint-data.json \
	--issue-file data/issue-data.json
WARN[0000] /Users/partisan/workshop/grantsgov/nava-simpler/analytics/docker-compose.yml: `version` is obsolete
[+] Creating 1/0
 ✔ Container grants-analytics-db  Running                                                                                                                                           0.0s
Warning: 'analytics' is an entry point defined in pyproject.toml, but it's not installed as a script. You may get improper `sys.argv[0]`.

The support to run uninstalled scripts will be removed in a future release.

Run `poetry install` to resolve and get rid of this message.

Usage: analytics export db_export [OPTIONS]
Try 'analytics export db_export --help' for help.
╭─ Error ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ No such option: --owner                                                                                                                                                               │
╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
make: *** [sprint-db-data-import] Error 2
```

**Comment by acouch** [comment]
Missing the issue data export, unless you plan to-do another PR for that one.

File: analytics/src/analytics/cli.py [hunk 3]
```diff
@@ -149,6 +149,30 @@ def test_connection() -> None:
...
+    connection = engine.connect()
+
+    # get data and load from JSON
+    task_data = DeliverableTasks.load_from_json_files(
+        sprint_file=sprint_file,
+        issue_file=issue_file,
+    )
+
+    BaseDataset.to_sql(
+        output_table=task_data,
```
Comment (acouch) [inline analytics/src/analytics/cli.py]: The `output_table` argument is a string for the table name. Can you recheck this locally? Did you have a working version here?

File: analytics/Makefile (hunk 2 above)
Comment (aplybeah) [reply analytics/Makefile]: I'm also thinking `import`

File: analytics/src/analytics/cli.py (hunk 3 above)
Comment (aplybeah) [reply analytics/src/analytics/cli.py]: @acouch it worked in the sense that the command ran locally, but I haven't been able to find the data in the database

File: analytics/poetry.lock
Comment (acouch) [inline analytics/poetry.lock:1]: Make sure to add an update to the pyproject.toml if you are updating the poetry.lock file.

File: analytics/src/analytics/cli.py (hunk 3 above)
Comment (acouch) [reply analytics/src/analytics/cli.py]: Try instantiating the class, then applying the .to_sql method :

```python

  deliverables = DeliverableTasks.load_from_json_files(
    sprint_file=sprint_file,
    issue_file=issue_file,
  )

  deliverables.to_sql(
    output_table="github_project_data",
    engine=connection,
    replace_table=True
  )
```

**Comment by aplybeah** [comment]
Expected Failures:
```
FAILED tests/integrations/test_slack.py::test_fetch_slack_channels - TypeError: 'NoneType' object is not subscriptable
FAILED tests/integrations/test_slack.py::test_upload_files_to_slack_channel - TypeError: 'NoneType' object is not subscriptable
```
The tests expect slack auth. Turning the validators off ended up interrupting the full suite

**Review by coilysiren — APPROVED** [review]
👍🏼! But you probably want to follow-up to disable those failing tests

File: analytics/Makefile (hunk 2 above)
Comment (acouch) [reply analytics/Makefile]: Please update with import. Please also create a separate import app: 

```python
import_app = typer.Typer()
...
app.add_typer(import_app, name="import", help="Import data into the database")

```

File: analytics/settings.toml [hunk 4]
```diff
@@ -1,4 +1,4 @@
 POSTGRES_NAME = "app"
-POSTGRES_HOST = "0.0.0.0"
+POSTGRES_HOST = "grants-analytics-db"
 POSTGRES_USER = "app"
 POSTGRES_PORT = 5432
```
Comment (acouch) [inline analytics/settings.toml:4]: I believe you would need a password here as well, can you confirm?

Comment (aplybeah) [reply analytics/settings.toml:4]: Theres a `secrets.toml` that's in the `gitignore` that has the password

**Review by coilysiren — APPROVED** [review]
👍🏼 !

**Comment by acouch** [comment]
Thanks for the updates. Can you add the `SprintBoard` data as well? https://github.com/navapbc/simpler-grants-gov/pull/84/files#diff-9ebe0c30e3ac3922cd1dd64b982265f1732a5d801148e6cdbf7a3ff5daa399a6R12

**Review by acouch — APPROVED** [review]

**Pull request merged**